wurst/boss/Minion.wurst

```diff
@@ -0,0 +1,105 @@
+package Minion
+import public EscaperSensor
+import Assets
+import Escaper
+import CustomBar
+import ClosureForGroups
+import Fx
+
+public constant COLL_SIZE_MINION = 0.
+public constant ATTACK_DISTANCE = 70.
+public constant ATTACK_RANGE = 120.
+public constant ATTACK_DAMAGE = 100.
+
+class MinionData
+    UnitEntity entity
+    
+    construct(UnitEntity k)
+        entity = k
+
+public class Minion extends EscaperSensorSetup
+    private Escaper target
+    real maxHp = 40.
+    real currentHP = 40.
+    CustomBar hpBar
+    boolean attacking = false
+
+    construct(vec2 pos, player owner, angle facing)
+        super(pos.toVec3(), owner, getRegionData(pos).getTypeSet().minionId, facing, COLL_SIZE_MINION, COLL_SIZE_MINION + 16)
+        setup()
+
+    construct(vec2 pos, player owner)
+        super(pos.toVec3(), owner, getRegionData(pos).getTypeSet().minionId, COLL_SIZE_MINION, COLL_SIZE_MINION + 16)
+        setup()
+
+    function setup()
+        selectTarget()
+        sleeps = false
+        hpBar = new CustomBar(pos, "|", 40, 6)
+        hpBar.setForeground(colorA(200,255,125,255))
+        
+    function onDamage(Escaper escaper, real dmg)
+        currentHP -= dmg
+        if currentHP < 1
+            actor.kill()
+            setup.abortOrder()
+            hpBar.fadeOut(1.5, true, 0, 0)
+            sleeps = true
+            getTimer()..start(3, function Minion.removeAfterDeath)..setData(new MinionData(this) castTo int)
+
+    override function update()
+        super.update()
+        hpBar.setPosXY(pos.x - 50, pos.y + 50)
+        hpBar.setValue(currentHP / maxHp * 100.0)
+        if target == null or target.alive != true or target.getCurrentRegion() != this.getCurrentRegion()
+            selectTarget()
```

Comment:
I think it is not a good idea to check for targets every tick. 
You could instead increase collisionsize and wake the minion up using onSensor (from the class Minion extends)
Or you add a buff to the minion that checks for targets - buffs are only updated 4 times per second.